packages/shared_preferences/shared_preferences/test/devtools_extension_test.dart

```diff
@@ -0,0 +1,324 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:shared_preferences/src/devtools_extension.dart';
+import 'package:shared_preferences_platform_interface/shared_preferences_async_platform_interface.dart';
+
+import 'shared_preferences_async_test.dart';
+
+typedef _Event = (String eventKind, Map<String, Object?> eventData);
+
+class _FakePostEvent {
+  final List<_Event> eventLog = <_Event>[];
+
+  void call(
+    String eventKind,
+    Map<String, Object?> eventData,
+  ) {
+    eventLog.add((eventKind, eventData));
+  }
+}
+
+void main() {
+  group('DevtoolsExtension', () {
+    late SharedPreferencesAsync asyncPreferences;
+    late _FakePostEvent fakePostEvent;
+    late DevtoolsExtension extension;
+
+    setUp(() {
+      SharedPreferencesAsyncPlatform.instance = FakeSharedPreferencesAsync();
+      asyncPreferences = SharedPreferencesAsync();
+      fakePostEvent = _FakePostEvent();
+      extension = DevtoolsExtension(fakePostEvent.call);
+    });
+
+    test('should request all keys', () async {
+      SharedPreferences.setMockInitialValues(<String, Object>{
+        'key1': 1,
+        'key2': true,
+      });
+      await asyncPreferences.setBool('key3', true);
+      await asyncPreferences.setInt('key4', 1);
+
+      await extension.requestAllKeys();
+
+      expect(fakePostEvent.eventLog.length, equals(1));
+      final (
+        String eventKind,
+        Map<String, Object?> eventData,
+      ) = fakePostEvent.eventLog.first;
+      expect(
+        eventKind,
+        equals('shared_preferences:all_keys'),
+      );
+      expect(
+        eventData,
+        equals(<String, List<String>>{
+          'asyncKeys': <String>['key3', 'key4'],
+          'legacyKeys': <String>['key1', 'key2'],
+        }),
+      );
+    });
+
+    test('should request value from async api', () async {
```

Comment:
for this test case and the test below, can you include what type the value is in the test name so that the tests are easy to differentiate (like you did for the async api test below)? Also, it looks like we are only testing bool and int values for the legacy API. Should we be testing every type like you do for the async api below?

Reply:
Since we're using the `get` method in the `requestValue` implementation, which returns a `Object?` I thought that doing a test for each type would be redundant. But I see the value of doing a test for each type, we could check if the `'kind'` value is being set accordingly. I'll create a helper function and write tests for all types.